src/cassio/table/mixins.py

```diff
@@ -584,6 +598,11 @@ def _schema_da(self) -> List[ColumnSpecType]:
             ("vector", f"VECTOR<FLOAT,{self.vector_dimension}>")
         ]
 
+    async def _aschema_da(self) -> List[ColumnSpecType]:
+        if inspect.isawaitable(self.vector_dimension):
+            self.vector_dimension = await self.vector_dimension
```

Comment:
This in-place overwrite is all happening at the (async closest equivalent of) constructor-time, right?

If for some reason it's not, it may be worth to split the two attributes to avoid any risk of mutability-induced confusion (?) but not so important, eh

Reply:
Yes. The idea is to be as close to the constructor as possible. So at he first invocation of _achema_da.
Also this way, we ensure that vector_dimension is only awaited once.